rpc/gnmi/gnmi.proto

```diff
@@ -129,6 +149,36 @@ message Error {
   google.protobuf.Any data = 3;   // Optional additional information.
 }
 
+
+// Decimal64 is used to encode floating-point decimal number. The value
+// expressed as a set of digits with the precision specifying the number
+// of digits following the decimal point in the digit set.
+message Decimal64 {
+  uint64 digits = 1;      // Set of digits.
+  uint32 precision = 2;    // Number of digits following the decimal point.
+}
+
+// ScalarArray is used to encode a mixed-type array of values.
+message ScalarArray {
+  // The set of elements contained within the array.
+  repeated ScalarArrayElement element = 1;
+}
+
+// ScalarArrayElement specifies an individual element of a scalar array.
+message ScalarArrayElement {
+  // One of the fields within the value oneof may be specified
+  // indicating an element included within the ScalarArray.
+  oneof value {
+    string string_val = 1;      // String value.
+    int64 int_val = 2;          // Integer value.
+    uint64 uint_val = 3;        // Unsigned integer value.
+    bool bool_val = 4;          // Boolean value.
+    bytes bytes_val = 5;        // Arbitrary byte sequence value.
+    float float_val = 6;        // Floating-point value.
+    Decimal64 decimal_val = 7;  // Decimal value.
```

Comment:
In YANG we can't have a leaf-list that contains a leaf-list so I omitted this case.